src/torchjd/autojac/mtl_backward.py

```diff
@@ -44,12 +126,12 @@ def mtl_backward(
     :param losses: The task losses. The Jacobian matrix will have one row per loss.
     :param features: The last shared representation used for all tasks, as given by the feature
         extractor. Should be non-empty.
+    :param A: Aggregator used to reduce the Jacobian into a vector.
     :param tasks_params: The parameters of each task-specific head. Their ``requires_grad`` flags
-        must be set to ``True``.
+        must be set to ``True``. Defaults to ``None``.
```

Comment:
We need to specify what happens if `None`, something like ``` If `None`, the tasks params are deduced to be the parameters that do not appearing the computation graph of all losses ```

Reply:
We accumulate in that case. This is why I suggest that we have a function that will do exactly the same, but will aggregate the parameters shared by any group of losses together (all subsets of losses).

Reply:
actually no, there is a potential for error here, and we should add a test.
Suppose we do something like
```python
a = tensor([1.], requires_grad=True)
b = tensor([1.], requires_grad=True)
x = 2 * a
y1 = 2 * b
y2 = 2 * x
y3 = 3 * x
```
Then no parameter is shared, also when we compute the gradient of y2 and y3 with respect to a, then we use the same graph of the gradient of x wrt a, so I think it is supposed we are supposed to `retain_graph` when doing the `Grad` transform. I think there is a way to deduce when we keep/drop the graph by going over the graph, but this is non-trivial.
A quick fix of this could be to provide an optional `retain_graph` flag to the `mtl_backward` function and use this for all, so if a user have a problem, they can change it, later we can have something a bit better in case we do not retain.

Reply:
> We accumulate in that case. This is why I suggest that we have a function that will do exactly the same, but will aggregate the parameters shared by any group of losses together (all subsets of losses).

I don't think this will be useful to a lot of people. This would basically be hierarchical MTL. Let's only add it if people need it.

Reply:
> actually no, there is a potential for error here, and we should add a test. Suppose we do something like
> 
> ```python
> a = tensor([1.], requires_grad=True)
> b = tensor([1.], requires_grad=True)
> x = 2 * a
> y1 = 2 * b
> y2 = 2 * x
> y3 = 3 * x
> ```
> 
> Then no parameter is shared, also when we compute the gradient of y2 and y3 with respect to a, then we use the same graph of the gradient of x wrt a, so I think it is supposed we are supposed to `retain_graph` when doing the `Grad` transform. I think there is a way to deduce when we keep/drop the graph by going over the graph, but this is non-trivial. A quick fix of this could be to provide an optional `retain_graph` flag to the `mtl_backward` function and use this for all, so if a user have a problem, they can change it, later we can have something a bit better in case we do not retain.

We already have a `retain_graph` param, so I'm not sure there's a problem here. However, I agree that we should add a test for this (unrelated to this PR).